packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -43,14 +43,31 @@ const String minSdkVersion = '19';
 const String targetSdkVersion = '33';
 const String ndkVersion = '23.1.7779620';
 
+
+// Update these when new major versions of Java are supported by Gradle.
+// Supported means Java <-> Gradle support.
+// Source of truth: https://docs.gradle.org/current/userguide/compatibility.html
+const String oneMajorVersionHigherJavaVersion = '20';
```

Comment:
```suggestion
const String oneMajorVersionHigherJavaVersion = '21';
```

per https://docs.gradle.org/current/userguide/compatibility.html#java

Reply:
Can you file a follow up bug for this?  I have a machine with java 20 that I use for flutter projects.

Reply:
https://github.com/flutter/flutter/issues/135022